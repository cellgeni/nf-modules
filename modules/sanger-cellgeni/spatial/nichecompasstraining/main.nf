process SPATIAL_NICHECOMPASSTRAINING {
    tag "${meta.id}"
    label 'process_gpu'

    container "quay.io/cellgeni/nichecompass:0.3.0"

    input:
    tuple val(meta), path(h5ad, stageAs: "inputs/*")

    output:
    tuple val(meta), path("nichecompass_*"), emit: nichecompass_model
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    nichecompass_train_sample_integration.py \\
        --batches ${h5ad} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        nichecompass: \$(pip show nichecompass | grep Version | sed -e "s/Version: //g")
    END_VERSIONS
    """

    stub:
    def timestamp = new Date().format("yyyyMMdd_HHmmss")
    """
    mkdir -p "nichecompass_${timestamp}"
    mkdir -p "nichecompass_${timestamp}/artifacts/figures"
    mkdir -p "nichecompass_${timestamp}/artifacts/models"
    mkdir -p "nichecompass_${timestamp}/data"
    touch "nichecompass_${timestamp}/artifacts/models/model.h5ad"
    touch "nichecompass_${timestamp}/train.log"
    touch "nichecompass_${timestamp}/timestamp.txt"

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        nichecompass: \$(pip show nichecompass | grep Version | sed -e "s/Version: //g")
    END_VERSIONS
    """
}
