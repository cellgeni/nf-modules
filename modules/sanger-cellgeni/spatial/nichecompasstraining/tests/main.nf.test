nextflow_process {

    name "Test Process SPATIAL_NICHECOMPASSTRAINING"
    script "../main.nf"
    process "SPATIAL_NICHECOMPASSTRAINING"

    tag "modules"
    tag "modules_sanger-cellgeni"
    tag "spatial"
    tag "spatial/nichecompasstraining"

    test("nichecompass_train - 3 samples") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test_20250829' ], // meta map
                    file(params.modules_testdata_base_path + 'nichecompass/training/100/*.h5ad', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("nichecompass_train - 3 samples - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test_20250829' ], // meta map
                    file(params.modules_testdata_base_path + 'nichecompass/training/100/*.h5ad', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
