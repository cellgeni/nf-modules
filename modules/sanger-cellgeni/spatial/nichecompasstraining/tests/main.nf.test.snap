{
    "mouse - 3 - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            [
                                [
                                    
                                ],
                                [
                                    "model.h5ad:md5,d41d8cd98f00b204e9800998ecf8427e"
                                ]
                            ],
                            [
                                
                            ],
                            "timestamp.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "train.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,0db64c3d3e9bc7683150a2079488147e"
                ],
                "nichecompass_model": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            [
                                [
                                    
                                ],
                                [
                                    "model.h5ad:md5,d41d8cd98f00b204e9800998ecf8427e"
                                ]
                            ],
                            [
                                
                            ],
                            "timestamp.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "train.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0db64c3d3e9bc7683150a2079488147e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-29T13:37:29.916405524"
    },
    "mouse - 3": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                [
                                    
                                ],
                                [
                                    "attr.pkl:md5,c4913d8bcf14b080e0eb941615a113f2",
                                    "model_params.pt:md5,200931e29ad7895613961389925e3cf0",
                                    "trained.h5ad:md5,5c4a0a1675ef673550eb5c8dc4aeade8",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_human.csv:md5,56d65fdb9c0e969a6270aeaabba75637",
                                    "nichenet_lr_network_v2_human.csv:md5,fd95490c94b0261cd92ebb70bda1c17f",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,e26aba1e348c9069fc7930b46a792574",
                            "timestamp.txt:md5,edd95ba4025daeaf81ec054f94ffef11",
                            "train.log:md5,9c4f9fbae5b4afbc4b52bff51f8861e5"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,0db64c3d3e9bc7683150a2079488147e"
                ],
                "nichecompass_model": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            [
                                [
                                    
                                ],
                                [
                                    "attr.pkl:md5,c4913d8bcf14b080e0eb941615a113f2",
                                    "model_params.pt:md5,200931e29ad7895613961389925e3cf0",
                                    "trained.h5ad:md5,5c4a0a1675ef673550eb5c8dc4aeade8",
                                    "var_names.csv:md5,d6db17b25f7a7847eac85d7b5a3552bc"
                                ]
                            ],
                            [
                                [
                                    "gencode.vM25.chr_patch_hapl_scaff.annotation.gtf.gz:md5,c77cbf3cad15af10b6174d4b77c8246f",
                                    "human_mouse_gene_orthologs.csv:md5,b01557bcc3647653d0b323e17fe1e93c"
                                ],
                                [
                                    [
                                        "human_metabolite_enzymes.tsv:md5,242269afd818d1ff93f190271065526c",
                                        "human_metabolite_sensors.tsv:md5,5a02dd9124c603d62e4944526f8b544c",
                                        "mouse_metabolite_enzymes.tsv:md5,f2a48c14af3f5174ae78cf390cbfbe0e",
                                        "mouse_metabolite_sensors.tsv:md5,1987eb7777861838ac8b6224732d88b8"
                                    ],
                                    "nichenet_ligand_target_matrix_v2_human.csv:md5,56d65fdb9c0e969a6270aeaabba75637",
                                    "nichenet_lr_network_v2_human.csv:md5,fd95490c94b0261cd92ebb70bda1c17f",
                                    "omnipath_lr_network.csv:md5,03487d78e4fe071fa42832be0bbdbf8d"
                                ]
                            ],
                            "run_config.json:md5,e26aba1e348c9069fc7930b46a792574",
                            "timestamp.txt:md5,edd95ba4025daeaf81ec054f94ffef11",
                            "train.log:md5,9c4f9fbae5b4afbc4b52bff51f8861e5"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0db64c3d3e9bc7683150a2079488147e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-29T13:34:35.4503056"
    }
}